core/templates/services/svg-sanitizer.service.ts

```diff
@@ -154,48 +154,72 @@ export class SvgSanitizerService {
     return { tags: invalidTags, attrs: invalidAttrs };
   }
 
+  // SVG is considered an XMLDocument and DOMParser's parseFromString method
+  // returns either a HTMLDocument or XMLDocument. Visit here for more info:
+  // https://developer.mozilla.org/en-US/docs/Web/API/DOMParser/parseFromString
+  // Document is the interface that both HTMLDocument and XMLDocument inherit
+  // properties from.
+  getSvgFromDataUri(dataURI: string): Document {
+    // Convert base64/URLEncoded data component to raw binary data
+    // held in a string.
+    let svgString = atob(dataURI.split(',')[1]);
+    let domParser = new DOMParser();
+    return domParser.parseFromString(svgString, 'image/svg+xml');
+  }
+
+  removeAllInvalidTagsAndAttributes(svgDataURI: string): string {
+    // We are removing the attributes which are currently not in
+    // the allowlist of valid attributes. The allowlist is based on
+    // the list of tags and attributes specified in this project:
+    // https://github.com/cure53/DOMPurify
+    // Complete list is present at 'assets/constants.ts'.
+    let svg = this.getSvgFromDataUri(svgDataURI);
+    let invalidTagsAndAttributes = this._getInvalidSvgTagsAndAttrs(svg);
+    let tagsToBeRemoved = invalidTagsAndAttributes.tags;
+    let attrsToBeRemoved = invalidTagsAndAttributes.attrs;
+    svg.querySelectorAll('*').forEach((node) => {
+      let nodeTagName: string = node.tagName.toLowerCase();
+      if (tagsToBeRemoved.indexOf(nodeTagName) !== -1) {
+        node.parentNode?.removeChild(node);
```

Comment:
Is it possible for there to be no parentNode (e.g. the top-level node is wrong)? Please add a test case for this.

Reply:
The top-level node would always be the <svg></svg> tag itself. I think a case with invalid top-level node wouldn't arise since misspelled svg tag won't be visible as an image while uploading.

Reply:
Ya I mean that it won't be uploaded since it wouldn't be classified as an image file. It would look something like this: 

https://user-images.githubusercontent.com/89375125/157314946-db30d9ba-be4c-43fc-b06d-b82510e0792b.mp4

Reply:
OK, maybe add a comment to explain why the parent tag will always exist then.

But in the video you show, shouldn't we show an error message to tell them that the SVG is invalid? Failing silently is not great from a UX perspective.

Reply:
I will add a comment for sure. 
Regarding the second part, I am not exactly sure of how we can achieve that. Maybe we would have to carry a check in ck-editor component; I'm not sure. Something same would happen if the user deliberately renamed a ".txt" or ".pdf" file to ".svg" file and tried to upload it.

Reply:
Don't you already have an existing function for checking whether a dataURI is a valid svg?

Reply:
Actually, the whole code executes only after a file change is detected. In this case, when an invalid "file" is uploaded as an image, it won't get accepted by the form and since no file is detected, no code will execute.

Reply:
That's interesting. We should still provide an error for that case, but I think you can do that in a separate PR since it's a different part of the flow and we need to get this one in.

Could you please file an issue for that with clear repro instructions, and point to it from here? That will address the second part, and perhaps we can look more deeply into it after this PR is completed. Thanks!

Reply:
Done! PTAL on [15102](https://github.com/oppia/oppia/issues/15102).